src/golang/lib/job/spec.go

```diff
@@ -122,11 +123,15 @@ type SystemMetricSpec struct {
 
 type ExtractSpec struct {
 	basePythonSpec
-	ConnectorName      integration.Service     `json:"connector_name"  yaml:"connector_name"`
-	ConnectorConfig    auth.Config             `json:"connector_config"  yaml:"connector_config"`
-	Parameters         connector.ExtractParams `json:"parameters"  yaml:"parameters"`
-	OutputContentPath  string                  `json:"output_content_path"  yaml:"output_content_path"`
-	OutputMetadataPath string                  `json:"output_metadata_path"  yaml:"output_metadata_path"`
+	ConnectorName   integration.Service     `json:"connector_name"  yaml:"connector_name"`
+	ConnectorConfig auth.Config             `json:"connector_config"  yaml:"connector_config"`
+	Parameters      connector.ExtractParams `json:"parameters"  yaml:"parameters"`
+
+	// Inputs can only be parameters.
+	InputContentPaths  []string `json:"input_content_paths" yaml:"input_content_paths"`
```

Comment:
we really need a task to make all paths a part of some struct like `Artifact` and pass the struct around, instead of n dict / lists

Reply:
Created https://linear.app/aqueducthq/issue/ENG-1289/pass-an-artifact-interface-around-the-orchestration-engine to follow up next sprint